src/main/java/de/tum/in/www1/artemis/service/ExerciseService.java

```diff
@@ -445,6 +452,18 @@ public void delete(long exerciseId, boolean deleteStudentReposBuildPlans, boolea
         }
         else {
             exerciseRepository.delete(exercise);
+            // delete text assessment knowledge if exercise is of type TextExercise and if no other exercise uses same knowledge
+            if (exercise instanceof TextExercise) {
+                log.info("Deleting knowledge of exercise");
```

Comment:
Do we need the log.info here? when deleting a programming exercise, we are also not logging any information.